lib/pengine/complex.c

```diff
@@ -596,16 +596,15 @@ pe__unpack_resource(xmlNode *xml_obj, pe_resource_t **rsc,
 
     if (id == NULL) {
         pe_err("Must specify id tag in <resource>");
-        return FALSE;
+        return pcmk_rc_schema_validation;
 
     } else if (rsc == NULL) {
         pe_err("Nowhere to unpack resource into");
-        return FALSE;
-
+        return EINVAL;
     }
 
     if (unpack_template(xml_obj, &expanded_xml, data_set) == FALSE) {
-        return FALSE;
+        return pcmk_rc_unpack_error;
```

Comment:
`unpack_template()` can fail due to a schema error or due to certain XML elements not found
* Missing object ID
* Improper reference to self
* No resources configured
* Named template does not exist in XML

A case could be made that nonexistent elements are still `pcmk_rc_schema_validation` errors, since other elements refer to or imply the existence of these nonexistent elements.

When `unpack_template()` is updated to return a standard Pacemaker return code later, we can just return its RC.

Reply:
Yep